controllers/dataprotection/backup_controller.go

```diff
@@ -752,7 +751,10 @@ func PatchBackupObjectMeta(
 	// TODO(ldm): we should remove this dependency of cluster in the future
 	cluster := getCluster(request.Ctx, request.Client, targetPod)
 	if cluster != nil {
-		if err := setClusterSnapshotAnnotation(request.Backup, cluster); err != nil {
+		if err := setClusterSnapshotAnnotation(request, cluster); err != nil {
+			return false, err
+		}
+		if err := setEncryptedSystemAccountsAnnotation(request, cluster); err != nil {
 			return false, err
 		}
 		if err := setConnectionPasswordAnnotation(request); err != nil {
```

Comment:
you can remove it.

Reply:
Should I remove the setClusterSnapshotAnnotation function? If I do, the compatibility processing will only consider restoring from old backups, and any new backups created will not have the ClusterSnapshotAnnotation.

Reply:
remove setConnectionPasswordAnnotation